### PR TITLE
Gradle Android devices

### PR DESCRIPTION
**Pull request by yschimke**
See https://developer.android.com/studio/preview/features?hl=hu

```
$ ./gradlew -Pandroid.experimental.androidTest.useUnifiedTestPlatform=true android-test:pixel2api30Check
```

Eventually we probably want this 
image

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -349,6 +349,6 @@ jobs:
         uses: reactivecircus/android-emulator-runner@v2
         with:
           api-level: ${{ matrix.api-level }}
-          script: ./gradlew -PandroidBuild :regression-test:connectedCheck
+          script: ./gradlew -PandroidBuild -Pandroid.experimental.androidTest.useUnifiedTestPlatform=true :regression-test:pixel2api29Check
```
Comment (yschimke) [inline .github/workflows/build.yml:352]: May revert this and just use for local testing below.

**Comment by yschimke** [comment]
Will simplify the README instructions later after a bit of testing first.

File: android-test/build.gradle.kts [hunk 2]
```diff
@@ -43,6 +43,31 @@ android {
     "README.txt",
     "org/bouncycastle/LICENSE"
   )
+
+  testOptions {
+    devices {
+      create<com.android.build.api.dsl.ManagedVirtualDevice>("pixel2api21") {
```
Comment (Goooler) [inline android-test/build.gradle.kts:49]: Advice to import `com.android.build.api.dsl.ManagedVirtualDevice`.

**Review by jj-crypto — APPROVED** [review]

**Review by swankjesse — APPROVED** [review]

**Review by jj-crypto — APPROVED** [review]

**Comment by yschimke** [comment]
Parking for now

**Pull request closed without merging**